deeplab/data.py

```diff
@@ -0,0 +1,684 @@
+        min_resize_scale (float): The minimum value the samples can be rescaled. Default: ``0.5``.
+        max_resize_scale (float): The maximum value the samples can be rescaled. Default: ``2.0``.
+        final_size (int): The final size of the image and target. Default: ``512``.
+        ignore_background (bool): If true, ignore the background class when calculating the training loss.
+            Default: ``true``.
+        **dataloader_kwargs (Dict[str, Any]): Additional settings for the dataloader (e.g. num_workers, etc.)
+    """
+    if version == 1:
+        warn_streaming_dataset_deprecation(old_version=version, new_version=2)
+        dataset = StreamingADE20k(remote=remote,
+                                  local=local,
+                                  split=split,
+                                  shuffle=shuffle,
+                                  base_size=base_size,
+                                  min_resize_scale=min_resize_scale,
+                                  max_resize_scale=max_resize_scale,
+                                  final_size=final_size,
+                                  batch_size=batch_size)
+    elif version == 2:
+
+        try:
+            import streaming
+        except ImportError as e:
+            raise MissingConditionalImportError(
+                extra_deps_group='streaming',
+                conda_package='mosaicml-streaming') from e
+
+        # Build the sets of transformations for ADE20k
+        both_transforms, image_transforms, target_transforms = build_ade20k_transformations(
+            split=split,
+            base_size=base_size,
+            min_resize_scale=min_resize_scale,
+            max_resize_scale=max_resize_scale,
+            final_size=final_size)
+
+        dataset = streaming.vision.ADE20K(remote=remote,
+                                          local=local,
+                                          split=split,
+                                          shuffle=shuffle,
+                                          both_transforms=both_transforms,
+                                          transform=image_transforms,
+                                          target_transform=target_transforms,
+                                          batch_size=batch_size)
+
+    else:
+        raise ValueError(f'Invalid streaming version: {version}')
+
+    dataloader = DataLoader(dataset=dataset,
+                            batch_size=batch_size,
+                            collate_fn=pil_image_collate,
+                            drop_last=drop_last,
+                            **dataloader_kwargs)
+    device_transform_fn = NormalizationFn(mean=IMAGENET_CHANNEL_MEAN,
+                                          std=IMAGENET_CHANNEL_STD,
+                                          ignore_background=ignore_background)
+
+    return DataSpec(dataloader=dataloader,
+                    device_transforms=device_transform_fn)
+
+
+def build_synthetic_ade20k_dataspec(
```

Comment:
Is this used anywhere? If not, should we delete it? I know synthetic datasets can be useful to measure model throughput, but I don't think we are setting this up in the benchmarks (yet).